### DISCUSSION TITLE
Front Matter support for comments

### DISCUSSION THREAD

**Discussion by code-not-found** [jpanther/congo#207] [Ideas]
Hi,

The theme has built in support for comments. I'm currently looking at enabling [utterances](utterances.se) and was able to achieve this really easily by creating a `comments.html` partial.

The comments are currently shown on all pages that use the `single.html` layout. However on some pages (terms, about, …) it doesn't always make sense to show the comments section. An elegant way to tackle this would be to add Front Matter support by defining a `article.showComments` parameter.

Let me know what you think.

**Comment by HyperTesto** [comment]
I was thinking the same. But I'd suggest to invert the parameter logic: something like `article.disableComments`

**Reply by code-not-found** [reply]
Nice to see this feature might interest others!

For the naming I borrowed inspiration from the [Hugo docs on front matter](https://gohugo.io/content-management/front-matter/#user-defined)

It mentions the `show_comments` user defined variable. As the other Congo variables seem to omit the underscore, I did the same in above example.

**Comment by jpanther** [comment]
I think this would be a useful addition. `showComments` would certainly be in line with the naming of the existing theme parameters. I would also split out `article` and `list` to be their own parameter. It could then inherit the default from the global config and be overridden at a per-page level.
